qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -495,3 +495,81 @@ def _square_real_number() -> SquareRealNumber:
 
 
 _SQUARE_REAL_NUMBER_DOC = BloqDocSpec(bloq_cls=SquareRealNumber, examples=[_square_real_number])
+
+
+@frozen
+class InvertRealNumber(Bloq):
+    r"""Invert a fixed-point representation of a real number.
+
+    Implements the unitary:
+    $$
+        |a\rangle|0\rangle \rightarrow |a\rangle|1/a\rangle
+    $$
+    where $a \ge 1$.
+
+    Args:
+        bitsize: Number of bits used to represent the number.
+        num_frac: Number of fraction bits in the number.
+
+    Registers:
+        a: `bitsize`-sized input register.
+        result: `bitsize`-sized output register.
+
+        References:
+        [Quantum Algorithms and Circuits for Scientific Computing](
+        https://arxiv.org/pdf/1511.08253). Section 2.1.
+    """
+
+    bitsize: int
+    num_frac: int
+
+    def __attrs_post_init__(self):
+        if (
+            isinstance(self.num_frac, int)
+            and isinstance(self.bitsize, int)
+            and self.num_frac >= self.bitsize
```

Comment:
num_frac > bitsize is a separate error right (I feel like this should be checked in __attrs_post_init__ of QFxp itself.). Also bitsize and num_frac are typed as ints in the constructor so I'm not sure what the first two checks are checking. I think you just want to check that num_frac != self.bitsize (as then it would be purely fractional and < 1) or am I getting confused? The number of integer bits is bitsize - num_frac.

Reply:
Thanks for the catch, this part is outdated and I will fix.